app/browser/api/ledger.js

```diff
@@ -0,0 +1,2187 @@
+        savePublisherOption(publisherKey, 'verified', result)
+      }
+    })
+  }
+
+  state = updatePublisherInfo(state)
+
+  return state
+}
+
+const enable = (state, paymentsEnabled) => {
+  if (paymentsEnabled && !getSetting(settings.PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED)) {
+    appActions.changeSetting(settings.PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED, true)
+  }
+
+  if (synopsis) {
+    return updatePublisherInfo(state)
+  }
+
+  if (!ledgerPublisher) {
+    ledgerPublisher = require('ledger-publisher')
+  }
+  synopsis = new (ledgerPublisher.Synopsis)()
+  const stateSynopsis = ledgerState.getSynopsis(state)
+
+  if (_internal.verboseP) {
+    console.log('\nstarting up ledger publisher integration')
+  }
+
+  if (stateSynopsis.isEmpty()) {
+    return initSynopsis(state)
+  }
+
+  try {
+    synopsis = new (ledgerPublisher.Synopsis)(stateSynopsis.toJS())
+  } catch (ex) {
+    console.error('synopsisPath parse error: ' + ex.toString())
+  }
+
+  state = initSynopsis(state)
+
+  // synopsis cleanup
+  underscore.keys(synopsis.publishers).forEach((publisher) => {
+    if (synopsis.publishers[publisher].faviconURL === null) {
+      delete synopsis.publishers[publisher].faviconURL
+    }
+  })
+
+  // change undefined include publishers to include publishers
+  state = ledgerState.enableUndefinedPublishers(state, stateSynopsis.get('publishers'))
+
+  return state
+}
+
+const pathName = (name) => {
+  const parts = path.parse(name)
+  return path.join(electron.app.getPath('userData'), parts.name + parts.ext)
+}
+
+const sufficientBalanceToReconcile = (state) => {
+  const balance = Number(ledgerState.getInfoProp(state, 'balance') || 0)
+  const unconfirmed = Number(ledgerState.getInfoProp(state, 'unconfirmed') || 0)
+  const btc = ledgerState.getInfoProp(state, 'btc')
+  return btc && (balance + unconfirmed > 0.9 * Number(btc))
+}
+
+const shouldShowNotificationReviewPublishers = () => {
+  const nextTime = getSetting(settings.PAYMENTS_NOTIFICATION_RECONCILE_SOON_TIMESTAMP)
+  return !nextTime || (new Date().getTime() > nextTime)
+}
+
+const shouldShowNotificationAddFunds = () => {
+  const nextTime = getSetting(settings.PAYMENTS_NOTIFICATION_ADD_FUNDS_TIMESTAMP)
+  return !nextTime || (new Date().getTime() > nextTime)
+}
+
+const showNotificationReviewPublishers = (nextTime) => {
+  appActions.changeSetting(settings.PAYMENTS_NOTIFICATION_RECONCILE_SOON_TIMESTAMP, nextTime)
+
+  appActions.showNotification({
+    greeting: locale.translation('updateHello'),
+    message: locale.translation('reconciliationNotification'),
+    buttons: [
+      {text: locale.translation('turnOffNotifications')},
+      {text: locale.translation('dismiss')},
+      {text: locale.translation('reviewSites'), className: 'primaryButton'}
+    ],
+    options: {
+      style: 'greetingStyle',
+      persist: false
+    }
+  })
+}
+
+const showNotificationAddFunds = () => {
+  const nextTime = new Date().getTime() + (3 * miliseconds.day)
```

Comment:
Another magic number (3) that we can pull up to the top as a constant 😄 👍

Reply:
done